From 0b02a6a7accc39c0f617858d4cc9710967bed0ad Mon Sep 17 00:00:00 2001
From: Balamurugan S <quic_bselvara@quicinc.com>
Date: Tue, 14 Mar 2023 22:07:14 +0530
Subject: [PATCH] ath12k: Map to cmem_remap structure when parsing
 ab->hw_params->cmem

Assign to cmem_remap structure when parsing cmem from hw params for
remapping the cmem memory. Initially this was assigned to ce_remap
since both structures had same variable did not cause any issues.
So, fix this by mapping to the cmem_remap structure.

Signed-off-by: Balamurugan S <quic_bselvara@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/ahb.c  | 4 ++--
 drivers/net/wireless/ath/ath12k/ce.h   | 5 -----
 drivers/net/wireless/ath/ath12k/core.h | 5 +++++
 drivers/net/wireless/ath/ath12k/hw.c   | 4 ++--
 drivers/net/wireless/ath/ath12k/hw.h   | 2 +-
 5 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/ahb.c b/drivers/net/wireless/ath/ath12k/ahb.c
index 13d5ce8..fbc998a 100644
--- a/drivers/net/wireless/ath/ath12k/ahb.c
+++ b/drivers/net/wireless/ath/ath12k/ahb.c
@@ -967,8 +967,8 @@ static int ath12k_ahb_probe(struct platform_device *pdev)
 		ab->ce_remap = true;
 		ab->ce_remap_base_addr = HAL_IPQ5332_CE_WFSS_REG_BASE;
 	}
-	if (ab->hw_params->cmem) {
-		const struct ce_remap *cmem = ab->hw_params->cmem;
+	if (ab->hw_params->cmem_remap) {
+		const struct cmem_remap *cmem = ab->hw_params->cmem_remap;
 		/*
 		 * In IPQ5332 CMEM region is outside WCSS block.
 		 * Allocate separate I/O remap to access CMEM address.
diff --git a/drivers/net/wireless/ath/ath12k/ce.h b/drivers/net/wireless/ath/ath12k/ce.h
index e951736..f709167 100644
--- a/drivers/net/wireless/ath/ath12k/ce.h
+++ b/drivers/net/wireless/ath/ath12k/ce.h
@@ -93,11 +93,6 @@ struct ce_remap {
 	u32 size;
 };
 
-struct ath12k_cmem {
-	u32 base;
-	u32 size;
-};
-
 struct ce_attr {
 	/* CE_ATTR_* values */
 	unsigned int flags;
diff --git a/drivers/net/wireless/ath/ath12k/core.h b/drivers/net/wireless/ath/ath12k/core.h
index 997ed75..27e61b3 100644
--- a/drivers/net/wireless/ath/ath12k/core.h
+++ b/drivers/net/wireless/ath/ath12k/core.h
@@ -999,6 +999,11 @@ struct ath12k_pdev {
 	struct mlo_timestamp timestamp;
 };
 
+struct cmem_remap {
+	u32 base;
+	u32 size;
+};
+
 #define BOARD_NAME_SIZE 100
 #define REGDB_NAME_SIZE 100
 
diff --git a/drivers/net/wireless/ath/ath12k/hw.c b/drivers/net/wireless/ath/ath12k/hw.c
index 117ce58..8665d55 100644
--- a/drivers/net/wireless/ath/ath12k/hw.c
+++ b/drivers/net/wireless/ath/ath12k/hw.c
@@ -1382,7 +1382,7 @@ const struct ce_remap ath12k_ce_remap_ipq5332 = {
 	.size = HAL_IPQ5332_CE_SIZE,
 };
 
-const struct ath12k_cmem ath12k_cmem_ipq5332 = {
+const struct cmem_remap ath12k_cmem_ipq5332 = {
 	.base = HAL_IPQ5332_CMEM_REG_BASE,
 	.size = HAL_IPQ5332_CMEM_SIZE,
 };
@@ -1687,7 +1687,7 @@ static const struct ath12k_hw_params ath12k_hw_params[] = {
 		.compact_rx_tlv = true,
 		.ce_ie_addr = &ath12k_ce_ie_addr_ipq5332,
 		.ce_remap = &ath12k_ce_remap_ipq5332,
-		.cmem = &ath12k_cmem_ipq5332,
+		.cmem_remap = &ath12k_cmem_ipq5332,
 		.send_platform_model = true,
 		.en_fwlog = false,
 	},
diff --git a/drivers/net/wireless/ath/ath12k/hw.h b/drivers/net/wireless/ath/ath12k/hw.h
index 279184e..3e9a577 100644
--- a/drivers/net/wireless/ath/ath12k/hw.h
+++ b/drivers/net/wireless/ath/ath12k/hw.h
@@ -221,7 +221,7 @@ struct ath12k_hw_params {
 	bool send_platform_model;
 	bool en_fwlog;
 	const struct ce_remap *ce_remap;
-	const struct cmem *cmem;
+	const struct cmem_remap *cmem_remap;
 	bool compact_rx_tlv;
 };
 
-- 
2.17.1

